vllm/engine/llm_engine.py

```diff
@@ -1246,33 +1266,35 @@ def _process_sequence_group_outputs(
 
         return
 
-    def _process_model_outputs(self, ctx: SchedulerContext) -> None:
-        """Apply the model output to the sequences in the scheduled seq groups.
+    def _process_model_outputs(self,
+                               ctx: SchedulerContext,
+                               request_id: Optional[str] = None) -> None:
+        """Apply the model output to the sequences in the scheduled seq groups
+        and return responses.
 
-        virtual_engine: The engine id to operate on
-        
-        is_async: Indicates whether this postprocessor runs in 
-            parallel with the GPU forward pass and is processing 
-            tokens from the previous step. If this is true, then
-            no tokens need to be appended since it is already done
-            externally (before the next schedule() call)
+        ctx: The virtual engine context to work on
+        request_id: If provided, then only this request is going to be processed
         
-        sampler_output: Used with multi-step execution to provide 
-            sampler_output of each step
-        is_last_output: Used with multi-step execution to indicate
-            the last step (of each multi-step group)
-            
-        Returns RequestOutputs that can be returned to the client.
         """
         now = time.time()
 
         if len(ctx.output_queue) == 0:
             return None
 
         # Get pending async postprocessor
-        (outputs, seq_group_metadata_list, scheduler_outputs, is_async,
-         is_last_step) = ctx.output_queue.popleft()
-        assert outputs is not None
+        if request_id:
+            # When we process only one request, no pop is required
+            # (since later we will process all of the rest)
+            output_data: OutputData = ctx.output_queue[0]
+        else:
+            output_data = ctx.output_queue.popleft()
+
+        outputs = output_data.outputs
```

Comment:
could make OutputData a NamedTuple and then it would be a bit simpler to destructure
```python
outputs, seq_group_metadata_list, ... = output_data
```

Reply:
Thanks for the suggestion, did not knew there is such thing, refactored the code to use NamedTuple.